spark-job-server-proposal.md

```diff
@@ -0,0 +1,61 @@
+# Adding Spark Job Server To Run Spark Jobs
+
+## Introduction
+
+This document provides design of spark job server which would trigger spark
+jobs either as scheduled or  batch jobs. Providing the user a ui for inputing
+the jar location, environment variables required to integrate with other apps,
+input parameters for spark job and files required.
+
+## Problem Statement
+
+Spark Applications can only run within OpenShift environment and users are
+required to do provide only source code which currently we use source to image
+to create a standard image which will be run against the spark cluster.
+Currently there is no way to schedule jobs or to run cron jobs. Also if
+users have a more complex setup where they would need infinispan-server or
+other servers they would be required to do a lot of setup.
+
+## Proposed Solution
+
+Apache Spark users will submit their spark job in the form of binary jar or py
+files. These files will require users to provide input as well as be triggered
+by spark-submit shell script. However there are other environment
+requirements that may be needed by users as well.
+
+The requirements to create, run, rerun, stop a particular spark job by one
+click of a button.
+
+With a simple html form users can provide the following input:
+* jar location / source git url
+* input parameters for spark job
+* files required to be mounted to run the job.
+* inject environment variables from services that this spark job would need
+to integrate with.
+
```

Comment:
i'd like to see more talk here about the workflow process for how the various components are expected to handle jobs. this can even just be a high-level pseudo-code walk through.